static/js/message_list_view.js

```diff
@@ -230,6 +230,22 @@ export class MessageListView {
         }
     }
 
+    set_calculated_message_container_variables(message_container) {
+        // The timestr of message_container can be outdated if locally
+        // echoed. When the server sends the new timestamp, though the
+        // timestamp gets updated in echo.js, the timestr does not.
+        // This updates the timestr which will be then be used to
+        // update the message_time during rerender.
```

Comment:
We can probably delete this comment in this second commit -- because we've now created a structure such that this bug won't recur by through creating this function.

Reply:
Sure updating just a minute.